docs/development/visualize/development-embedding-visualizations.asciidoc

```diff
@@ -1,89 +1,68 @@
-
-`container` should be a dom element to which visualization should be embedded
-`params` is a parameter object where the following properties can be defined:
-
-- `timeRange`: time range to pass to `<visualize>` directive
-- `uiState`: uiState to pass to `<visualize>` directive
-- `appState`: appState to pass to `<visualize>` directive
-- `showSpyPanel`: showSpyPanel property to pass to `<visualize>` directive
+There are two different methods you can use to insert a visualization in your page.
 
+To display an already saved visualization, use the `VisualizeLoader`.
+To reuse an existing visualization implementation for a more custom purpose,
+use the Angular `<visualization>` directive instead.
 
-==== `<visualize>` directive
-The `<visualize>` directive takes care of loading data, parsing data, rendering the editor
-(if the Visualization is in edit mode) and rendering the visualization.
-The directive takes a savedVis object for its configuration.
-It is the easiest way to add visualization to your page under the assumption that
-the visualization you are trying to display is saved in kibana.
-If that is not the case, take a look at using `<visualization>` directive.
-
-The simplest way is to just pass `saved-id` to `<visualize>`:
-
-`<visualize saved-id="'447d2930-9eb2-11e7-a956-5558df96e706'"></visualize>`
+==== VisualizeLoader
 
-For the above to work with time based visualizations time picker must be present (enabled) on the page. For scenarios
-where timepicker is not available time range can be passed in as additional parameter:
+The `VisualizeLoader` class is the easiest way to embed a visualization into your plugin.
+It will take care of loading the data and rendering the visualization.
 
-`<visualize saved-id="'447d2930-9eb2-11e7-a956-5558df96e706'"
-time-range="{ max: '2017-09-21T21:59:59.999Z', min: '2017-09-18T22:00:00.000Z' }"></visualize>`
+To get an instance of the loader, do the following:
 
-Once <visualize> is done rendering the element will emit `renderComplete` event.
+["source","js"]
+-----------
+import { getVisualizeLoader } from 'ui/visualize/loader';
 
-When more control is required over the visualization you may prefer to load the saved object yourself and then pass it
-to `<visualize>`
+getVisualizeLoader().then((loader) => {
+  // You now have access to the loader
+});
+-----------
 
-`<visualize saved-obj='savedVis' app-state='appState' ui-state='uiState' editor-mode='false'></visualize>` where
+The loader exposes the following methods:
 
-`savedVis` is an instance of savedVisualization object, which can be retrieved using `savedVisualizations` service
-which is explained later in this documentation.
+- `getVisualizationList()`: which returns promise which gets resolved with a list of saved visualizations
+- `embedVisualizationWithId(container, savedId, params)`: which embeds visualization by id
+- `embedVisualizationWithSavedObject(container, savedObject, params)`: which embeds visualization from saved object
```

Comment:
i think the parts describing the params of <visualize> directive should be moved here as well (not just deleted) ... 
like time-range, showSpyPanel, uiState and appState should probably have some explanation, as well as newly introduced class and data attributes that can be passed in.

Reply:
ah sorry, i missed the part where you reference the source code and that all parameters are actually described there. ignore my above comment